src/ui/src/builder/settings/BuilderFieldsToolsWithMcp.vue

```diff
@@ -0,0 +1,494 @@
+								@update:model-value="toggleTool(tool, $event)"
+							/>
+							<div
+								class="functionInfo"
+								@click="toggleTool(tool, !isToolSelected(tool))"
+							>
+								<div class="functionName">
+									{{ tool.functionName }}
+								</div>
+								<div
+									v-if="tool.function?.description"
+									class="functionDescription"
+								>
+									{{ tool.function.description }}
+								</div>
+							</div>
+						</div>
+					</div>
+				</div>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script setup lang="ts">
+import { computed, inject, onMounted, ref, toRef } from "vue";
+import { Component } from "@/writerTypes";
+import { useComponentFieldViewModel } from "../useComponentFieldViewModel";
+import WdsButton from "@/wds/WdsButton.vue";
+import WdsIcon from "@/wds/WdsIcon.vue";
+import WdsCheckbox from "@/wds/WdsCheckbox.vue";
+import WdsSkeletonLoader from "@/wds/WdsSkeletonLoader.vue";
+import injectionKeys from "@/injectionKeys";
+import { useWriterApi } from "@/composables/useWriterApi";
+import type { WriterApiMcpTool } from "@/writerApi";
+
+type FunctionTool = {
+	type: "function";
+	description: string;
+	parameters: Record<
+		string,
+		{
+			type: string;
+			description: string;
+		}
+	>;
+};
+
+type GraphTool = {
+	type: "graph";
+	graph_ids: string[];
+};
+
+type WebSearchTool = {
+	type: "web_search";
+	include_domains?: string[];
+	exclude_domains?: string[];
+	include_raw_content?: boolean;
+};
+
+type McpToolFunction = {
+	description?: string;
+	parameters?: Record<string, unknown>;
+	name?: string;
+	[key: string]: unknown;
+};
+
+type McpToolType = {
+	type: "mcp";
+	appId: string;
+	functionName: string;
+	function: McpToolFunction;
+};
+
+type Tool = FunctionTool | GraphTool | WebSearchTool | McpToolType;
+
+const wf = inject(injectionKeys.core);
+
+const { writerApi } = useWriterApi();
+
+const mcpToolsData = ref<WriterApiMcpTool[]>([]);
+const mcpToolsLoading = ref(false);
+const mcpToolsError = ref<Error | undefined>(undefined);
+const expandedApps = ref<Set<string>>(new Set());
+
+type GroupedTool = {
+	functions: WriterApiMcpTool[];
+	logo?: string;
+};
+
+const groupedTools = computed(() => {
+	const grouped: Record<string, GroupedTool> = {};
+
+	for (const tool of mcpToolsData.value) {
+		const appName = tool.appName || "Unknown App";
+		if (!grouped[appName]) {
+			grouped[appName] = {
+				functions: [],
+				logo: tool.connector?.logo,
+			};
+		}
+		grouped[appName].functions.push(tool);
+	}
+	return grouped;
+});
+
+const selectedCountsByApp = computed(() => {
+	const counts: Record<string, number> = {};
+	for (const [appName, group] of Object.entries(groupedTools.value)) {
+		counts[appName] = group.functions.filter((tool) =>
+			isToolSelected(tool),
+		).length;
+	}
+	return counts;
+});
+
+function toggleApp(appName: string) {
+	if (expandedApps.value.has(appName)) {
+		expandedApps.value.delete(appName);
+	} else {
+		expandedApps.value.add(appName);
+	}
```

Comment:
Do we need something like this in vue.js @madeindjs ?
```suggestion
    let clonedApps = new Set(expandedApps)
	if (clonedApps.value.has(appName)) {
		clonedApps.value.delete(appName);
	} else {
		clonedApps.value.add(appName);
	}
	expandedApps = new Set([...clonedApps]);
```